### PR TITLE
Allow for clear functionality without generating a button

### PR DESCRIPTION
**Pull request by matatirosolutions**
This PR implements #3469 and addresses my desire to be able to generate static buttons to clear filters.

### CODE REVIEW

**Comment by djhvscf** [comment]
Please update the Readme with the documentation for GenerateClear

**Comment by matatirosolutions** [comment]
Documentation updated :-)

**Comment by wenzhixin** [comment]
@matatirosolutions Please update src file instead of dist file.

File: dist/extensions/filter-control/bootstrap-table-filter-control.js [hunk 1]
```diff
@@ -520,18 +521,25 @@
...
-            if (!$btnClear.length) {
-                $btnClear = $([
-                    '<button class="btn btn-default filter-show-clear" ',
-                    sprintf('type="button" title="%s">', this.options.formatClearFilters()),
-                    sprintf('<i class="%s %s"></i> ', this.options.iconsPrefix, this.options.icons.clear),
-                    '</button>'
-                ].join('')).appendTo($btnGroup);
-
-                $btnClear.off('click').on('click', $.proxy(this.clearFilterControl, this));
+            if(this.options.filterGenerateClear) {
```
Comment (wenzhixin) [inline dist/extensions/filter-control/bootstrap-table-filter-control.js]: `if (this.options.filterGenerateClear) {`

File: dist/extensions/filter-control/bootstrap-table-filter-control.js [hunk 2]
```diff
@@ -520,18 +521,25 @@
...
+                        sprintf('type="button" title="%s">', this.options.formatClearFilters()),
+                        sprintf('<i class="%s %s"></i> ', this.options.iconsPrefix, this.options.icons.clear),
+                        '</button>'
+                    ].join('')).appendTo($btnGroup);
+
+                    $btnClear.off('click').on('click', $.proxy(this.clearFilterControl, this));
+                }
+            } else {
+                var $btnClear = $('.filter-clear');
+                if($btnClear.length) {
```
Comment (wenzhixin) [inline dist/extensions/filter-control/bootstrap-table-filter-control.js]: `if ($btnClear.length) {`

**Comment by djhvscf** [comment]
Hi! We don't have any reply in that case we are going to close this PR. If you want, please create a new **PR** with this change against develop.

**Pull request closed without merging**